crates/vm/src/eval.rs

```diff
@@ -44,6 +44,55 @@ fn build_query_opt(q: QueryExpr) -> QueryExprOpt {
     QueryExprOpt { source, query: q.query }
 }
 
+pub(crate) fn optimize_query(q: CrudExpr) -> ExprOpt {
```

Comment:
Can you make this patch, just about removing `EnvDb`? So essentially this patch should just have code removed, but not added - i.e. all red, no green.